FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,60 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url, completion: { [weak self] result in
+			guard self != nil else { return }
+
+			var completionResult: FeedLoader.Result = .failure(Error.invalidData)
+			defer {
+				completion(completionResult)
+			}
+
+			do {
+				let (data, urlResponse) = try result.get()
+
+				guard urlResponse.statusCode == 200 else { throw Error.invalidData }
+
+				let feedResponse = try JSONDecoder().decode(FeedItemsResponse.self, from: data)
```

Comment:
Since a `JSONDecoder` error would also lead to `Error.invalidData`, it's possible to combine these two statements into a single one, to simplify our logic:
```suggestion
				guard urlResponse.statusCode == 200,
					  let feedResponse = try? JSONDecoder().decode(FeedItemsResponse.self, from: data) else {
					throw Error.invalidData
				}
```
That way, we can remove the `catch is DecodingError` pattern from our `catches`.

Reply:
Simplified and separated this logic into separate method in 1b9b955